fendermint/eth/api/src/mpool.rs

```diff
@@ -57,8 +118,22 @@ async fn tx_cache_clearing_loop(
                                 block: Some(block), ..
                             } = event.data
                             {
-                                let tx_hashes = collect_tx_hashes(&block, &chain_id);
-                                tx_cache.remove_many(&tx_hashes);
+                                let txs = collect_txs(&block, &chain_id);
+
+                                if txs.is_empty() {
+                                    continue;
+                                }
+
+                                let tx_hashes = txs.iter().map(|(h, _, _)| h);
+                                let tx_nonces = || txs.iter().map(|(_, s, n)| (s, *n));
+
+                                tx_cache.remove_many(tx_hashes);
+                                // First remove all transactions which have been in the block (could be multiple from the same sender).
+                                tx_buffer.remove_many(tx_nonces());
+                                // Then collect whatever is enabled on top of those, ie. anything that hasn't been included, but now can.
+                                let enabled = tx_buffer.remove_enabled(tx_nonces());
```

Comment:
`enabled` seems like a boolean, maybe use sth like `enabled_txs`

Reply:
Renaming to `unblocked_msgs`